docs/Toy/toy/compiler.py

```diff
@@ -124,20 +124,32 @@ def transform(
     DeadCodeElimination().apply(ctx, module_op)
     ReconcileUnrealizedCastsPass().apply(ctx, module_op)
 
-    DeadCodeElimination().apply(ctx, module_op)
-
     module_op.verify()
 
     if target == "riscv":
         return
 
+    CanonicalizePass().apply(ctx, module_op)
```

Comment:
optimises things that don't need register allocation, like constant folding etc

Reply:
```suggestion
    # Perform optimizations that don't depend on register allocation (e.g.: constant folding):
    CanonicalizePass().apply(ctx, module_op)
```